api/service/metrics/service.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/gorilla/mux"
+	msg_pb "github.com/harmony-one/harmony/api/proto/message"
+	"github.com/harmony-one/harmony/p2p"
+	libp2p_peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/rs/zerolog/log"
+)
+
+// Constants for metrics service.
+const (
+	ConnectionsNumberPush        int = 0
+	BlockHeightPush              int = 1
+	NodeBalancePush              int = 2
+	LastConsensusPush            int = 3
+	BlockRewardPush              int = 4
+	metricsServicePortDifference     = 2000
+)
+
+// Service is the struct for metrics service.
+type Service struct {
+	router          *mux.Router
+	BlsPublicKey    string
+	IP              string
+	Port            string
+	PushgatewayIP   string
+	PushgatewayPort string
+	GetNodeIDs      func() []libp2p_peer.ID
+	storage         *Storage
+	pusher          *push.Pusher
+	messageChan     chan *msg_pb.Message
+}
+
+// init vars for prometheus
+var (
+	curBlockHeight       = uint64(0)
+	curBlocks            = uint64(0)
+	curBalance           = big.NewInt(0)
+	curConnectionsNumber = 0
+	lastBlockReward      = big.NewInt(0)
+	lastConsensusTime    = int64(0)
+	metricsPush          = make(chan int)
+	blockHeightCounter   = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "block_height",
+		Help: "Get current block height.",
+	})
+	blocksAcceptedGauge = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "blocks_accepted",
+		Help: "Get accepted blocks.",
+	})
+	connectionsNumberGauge = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "connections_number",
+		Help: "Get current connections number for a node.",
+	})
+	nodeBalanceCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "node_balance",
+		Help: "Get current node balance.",
+	})
+	lastConsensusGauge = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "last_consensus",
+		Help: "Get last consensus time.",
+	})
+	blockRewardGauge = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "block_reward",
+		Help: "Get last block reward.",
+	})
+)
+
+// ConnectionsLog struct for connections stats for prometheus
+type ConnectionsLog struct {
+	Time              int
+	ConnectionsNumber int
+}
+
+// ConnectionsStatsHTTP struct for returning all connections logs
+type ConnectionsStatsHTTP struct {
+	ConnectionsLogs []ConnectionsLog
+}
+
+// New returns metrics service.
+func New(selfPeer *p2p.Peer, blsPublicKey, pushgatewayIP, pushgatewayPort string, GetNodeIDs func() []libp2p_peer.ID) *Service {
+	return &Service{
+		BlsPublicKey:    blsPublicKey,
+		IP:              selfPeer.IP,
+		Port:            selfPeer.Port,
+		PushgatewayIP:   pushgatewayIP,
+		PushgatewayPort: pushgatewayPort,
+		GetNodeIDs:      GetNodeIDs,
+	}
+}
+
+// StartService starts metrics service.
+func (s *Service) StartService() {
+	log.Info().Msg("Starting metrics service.")
```

Comment:
Please use utils.GetLogger() to get the log. This is applicable to all log here.